awx/ui_next/src/screens/Inventory/InventorySourceAdd/InventorySourceAdd.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useCallback, useEffect } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import { InventorySourcesAPI } from '@api';
+import useRequest from '@util/useRequest';
+import { Card } from '@patternfly/react-core';
+import { CardBody } from '@components/Card';
+import InventorySourceForm from '../shared/InventorySourceForm';
+
+function InventorySourceAdd() {
+  const history = useHistory();
+  const { id } = useParams();
+
+  const { error, request, result } = useRequest(
+    useCallback(async values => {
+      const { data } = await InventorySourcesAPI.create(values);
+      return data;
+    }, [])
+  );
+
+  useEffect(() => {
+    if (result) {
+      history.push(
+        `/inventories/inventory/${result.inventory}/sources/${result.id}/details`
+      );
+    }
+  }, [result, history]);
+
+  const handleSubmit = async form => {
+    const { credential, source_path, source_project, ...remainingForm } = form;
+
+    const sourcePath = {};
+    const sourceProject = {};
```

Comment:
When I select a project initially, the validation tells me "please select a value" as if nothing was chosen. This doesn't affect submission, but the validation should probably be checked here.